src/main/java/com/exactpro/th2/common/schema/factory/CommonFactory.java

```diff
@@ -344,34 +335,88 @@ public static CommonFactory createFromArguments(String... args) {
+            settings.setDictionaryTypesDir(calculatePath(parsedArgs.dictionariesDir, configs, DICTIONARY_TYPE_DIR_NAME));
+            settings.setDictionaryAliasesDir(calculatePath(parsedArgs.dictionariesDir, configs, DICTIONARY_ALIAS_DIR_NAME));
+            String oldDictionariesDir = parsedArgs.dictionariesDir;
             settings.setOldDictionariesDir(oldDictionariesDir == null ? (configs == null ? CONFIG_DEFAULT_PATH : Path.of(configs)) : Path.of(oldDictionariesDir));
 
+            String configurationProviderClassName = defaultIfNull(parsedArgs.configurationProviderClass, DEFAULT_CONFIGURATION_PROVIDER.getName());
+
+            var providerClass = (Class<? extends ConfigurationProvider>) Class.forName(configurationProviderClassName);
+            var factory = loadFactoryForProvider(providerClass);
+            ConfigurationProviderFactory configurationProviderFactory = factory.get();
+
+
+            ConfigurationProviderConfig configurationProviderConfig =
+                    parseAs(args, configurationProviderFactory.settings());
+
+            settings.setConfigurationProviderClass(providerClass);
+            settings.setConfigurationProviderConfig(configurationProviderConfig);
+
             return new CommonFactory(settings);
-        } catch (ParseException e) {
+        } catch (ClassNotFoundException e) {
+            LOGGER.error("Failed to find configuration provider class: " + e.getMessage());
             throw new IllegalArgumentException("Incorrect arguments " + Arrays.toString(args), e);
         }
     }
 
+    private static ConfigurationProviderConfig parseAs(String[] args, ConfigurationProviderConfig config) {
+        JCommander.newBuilder()
+                .addObject(config)
+                .addObject(new CommonFactoryArgs())
+                .acceptUnknownOptions(false)
+                .build()
+                .parse(args);
+        return config;
+    }
+    private static CommonFactoryArgs parseAsCommonConfig(String[] args) {
+        CommonFactoryArgs commonFactoryArgs = new CommonFactoryArgs();
+        JCommander.newBuilder()
+                .addObject(commonFactoryArgs)
+                .acceptUnknownOptions(true)
+                .build()
+                .parse(args);
+        return commonFactoryArgs;
+    }
+
+    private static ServiceLoader.@NotNull Provider<ConfigurationProviderFactory> loadFactoryForProvider(
+            Class<? extends ConfigurationProvider> providerClass) {
+
+        var optionalFactory = configurationProviderFactoryLoader.stream()
+                .filter(factory -> factory.get().getType().equals(providerClass))
+                .findFirst();
+        if (optionalFactory.isEmpty()) {
+            LOGGER.error("Cannot find provider {}", providerClass);
+            throw new IllegalArgumentException("Cannot find provider " + providerClass);
+        }
+        return optionalFactory.get();
```

Comment:
Can be replaced with `orElseThrow`
```suggestion
        return optionalFactory.orElseThrow(() -> {
            LOGGER.error("Cannot find provider {}", providerClass);
            return new IllegalArgumentException("Cannot find provider " + providerClass);
        });
```